### DISCUSSION TITLE
Reason for removal of GraphicsMagick and ImageMagick

### DISCUSSION THREAD

**Discussion by JanPretzel** [cantaloupe-project/cantaloupe#526] [Q&A]
Just wanted to ask if there is a specific reason, that GraphicsMagick and ImageMagick got removed in v5? I am asking because as both seem to solve a problem we are currently facing with v5, as described in #524.

**Comment by adolski** [comment]
There are several reasons:

1. Running as an external process as they were, IM/GM can only do one thing per invocation. To read dimensions, metadata, and then image data, for example, would require three opens & closes, which is inefficient and maybe very slow, depending on the configuration.

2. They don't support random access when reading from streams

3. They don't support robust metadata I/O--for example, reading raw EXIF/IIM is not possible and would require even more process invocations even if it were. They don't support injection of custom metadata when writing.

4. They are more code to maintain and support while adding no significant value.
